modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotMessageCounter.java

```diff
@@ -361,39 +353,35 @@ private void updateCoordinatorWithTimedOutSlots(Set<Map.Entry<String, Long>> slo
         }
 
         /**
-         * Local nodes safe-zone is sent to the coordinator. This is done to keep the safezone moving forward when
+         * Local nodes safe-zone is sent to the coordinator. This is done to keep the safe-zone moving forward when
          * there are no publishers in the local node.
          */
         private void updateCoordinatorWithCurrentSafezone() {
-            slotSubmitLoopSkipCount++;
-            if (slotSubmitLoopSkipCount == SLOT_SUBMIT_LOOP_SKIP_COUNT_THRESHOLD) {
-                //update current slot Deletion Safe Zone
-                try {
-
-                    long evaluatedSafeZone = currentSlotDeleteSafeZone;
 
-                    // If there are any slots pending submission to coordinator, we must lower the safe zone to their
-                    // starting point.
-
-                    // If we do not consider pending slots at this calculation, safe zone will fly up unexpectedly.
-                    for (Map.Entry<String, Slot> slotEntry : queueToSlotMap.entrySet()) {
-                        if (!checkMessageLimitReached(slotEntry.getValue())) {
-                            evaluatedSafeZone = Math.min(slotEntry.getValue().getStartMessageId(),
-                                    evaluatedSafeZone);
-                        }
-                    }
-
-                    if (log.isDebugEnabled()) {
-                        log.debug("Updating coordinator with local safe zone " + evaluatedSafeZone);
+            //update current slot Deletion Safe Zone
+            try {
+                long evaluatedSafeZone = currentSlotDeleteSafeZone;
+
+                // If there are any slots pending submission to coordinator, we must lower the safe zone to their
+                // starting point.
+                // If we do not consider pending slots at this calculation, safe zone will fly up unexpectedly.
+                for (Map.Entry<String, Slot> slotEntry : queueToSlotMap.entrySet()) {
+                    if (!checkMessageLimitReached(slotEntry.getValue())) {
+                        evaluatedSafeZone = Math.min(slotEntry.getValue().getStartMessageId(),
+                                evaluatedSafeZone);
                     }
+                }
 
-                    submitCurrentSafeZone(evaluatedSafeZone);
-                    currentSlotDeleteSafeZone = evaluatedSafeZone;
-                    slotSubmitLoopSkipCount = 0;
-                } catch (ConnectionException e) {
-                    log.error("Error while sending slot deletion safe zone update", e);
+                if (log.isDebugEnabled()) {
+                    log.debug("Updating coordinator with local safe zone " + evaluatedSafeZone);
                 }
+
+                submitCurrentSafeZone(evaluatedSafeZone);
+                currentSlotDeleteSafeZone = evaluatedSafeZone;
+            } catch (ConnectionException e) {
+                log.error("Error while sending slot deletion safe zone update", e);
```

Comment:
Since the error itself will anyway get printed, adding that will duplicate the message.